db-service/lib/cqn2sql.js

```diff
@@ -832,9 +832,9 @@ class CQN2SQLRenderer {
       case 'number': return `${val}` // REVISIT for HANA
       case 'object':
         if (val === null) return 'NULL'
-        if (val instanceof Date) return `'${val.toISOString()}'`
-        if (val instanceof Readable); // go on with default below
-        else if (Buffer.isBuffer(val)); // go on with default below
+        if (val instanceof Date) val = val.toJSON() // returns null if invalid
```

Comment:
When would an instance of Date be invalid?

Reply:
When someone would provide an invalid `Date` in their query.

```js
SELECT.from('Books').where(`createdAt <`, new Date('0000-00-00'))
```

This would throw an error when parsing it into an `ISO` string. Without any indication where the `Date` instance was created. Making it look like an database service error.

Reply:
Ok. But then, is it correct to silently have that running into another SQL error like 'comparing with NULL osing < operator is not supported'?

Reply:
Having the database throw the error would be a lot more clear in this instance. Another option would be to check whether the `Date` is valid and throw our own error. As currently the error thrown is really confusing.

```
Uncaught RangeError: Invalid time value
<database service stack traces>
```

Where a `SQL` error would at least include the original `SQL` statement that caused the error. With often additional information showing which operator / column was being compared with.